app/assets/javascripts/player_listeners.js

```diff
@@ -0,0 +1,495 @@
+/* 
+ * Copyright 2011-2024, The Trustees of Indiana University and Northwestern
+ *   University.  Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed
+ *   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ---  END LICENSE_HEADER BLOCK  ---
+*/
+
+let canvasIndex = -1;
+let currentSectionLabel = undefined;
+let listenersAdded = false;
```

Comment:
Is this used only for add to playlist listeners?  If so how about renaming this variable?
```suggestion
let addToPlaylistListenersAdded = false;
```

Reply:
Yes, that sounds good 👍